go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -685,66 +685,61 @@ func buildUpdateLogicalPlan(
 	hints *queryHints,
 ) (logicalPlan, error) {
 	upd := dmlOp.(*operators.Update)
-	rp := newRoutingParams(ctx, rb.Routing.OpCode())
-	rb.Routing.UpdateRoutingParams(ctx, rp)
-	edml := &engine.DML{
-		Query:             generateQuery(stmt),
-		TableNames:        []string{upd.VTable.Name.String()},
-		Vindexes:          upd.VTable.ColumnVindexes,
-		OwnedVindexQuery:  upd.OwnedVindexQuery,
-		RoutingParameters: rp,
+	var vindexes []*vindexes.ColumnVindex
+	vQuery := ""
+	if len(upd.ChangedVindexValues) > 0 {
+		vQuery = sqlparser.String(upd.OwnedVindexQuery)
+		vindexes = upd.Target.VTable.ColumnVindexes
+		if upd.OwnedVindexQuery.Limit != nil && len(upd.OwnedVindexQuery.OrderBy) == 0 {
+			return nil, vterrors.VT12001("Vindex update should have ORDER BY clause when using LIMIT")
```

Comment:
is this an unsupported feature or a wrong input from the user?

Reply:
unsupported currently.

Reply:
as the output is non-deterministic.

Reply:
got it